libs/utils/utils.js

```diff
@@ -195,24 +202,56 @@ export function getEnv(conf) {
   /* c8 ignore stop */
 }
 
-export function getLocale(locales, pathname = window.location.pathname) {
-  if (!locales) {
-    return { ietf: 'en-US', tk: 'hah7vzn.css', prefix: '' };
+function hydrateLocale(locales, key) {
+  const locale = locales[key];
+
+  const buildExpandedLocale = (localeData, localeKey) => ({
+    ...localeData,
+    prefix: localeKey ? `/${localeKey}` : '',
+    region: localeData.region || localeKey.split('_')[0] || 'us',
+  });
+
+  const isBaseLocale = !('base' in locale);
+  if (isBaseLocale) {
+    const childLocaleEntries = Object.entries(locales)
+      .filter(([, childLocale]) => childLocale.base === key);
+
+    const hydratedChildren = childLocaleEntries.reduce((acc, [childKey, childLocale]) => {
+      const mergedLocale = { ...locale, ...childLocale, base: childLocale.base };
+      acc[childKey] = buildExpandedLocale(mergedLocale, childKey);
+      return acc;
+    }, {});
+
+    const hydratedBase = buildExpandedLocale(locale, key);
+    return { ...hydratedBase, regions: hydratedChildren };
   }
+
+  const hasValidBase = (locale.base || locale.base === '') && locales[locale.base];
+  if (hasValidBase) {
+    const baseLocale = locales[locale.base];
+    const mergedLocale = { ...baseLocale, ...locale };
+    return buildExpandedLocale(mergedLocale, key);
+  }
+
+  return { ...locale };
+}
```

Comment:
Skips a few one-time variables, simplified check, chaining
```suggestion
function hydrateLocale(locales, key) {
  const locale = locales[key];

  const buildExpandedLocale = (localeData, localeKey) => ({
    ...localeData,
    prefix: localeKey ? `/${localeKey}` : '',
    region: localeData.region || localeKey.split('_')[0] || 'us',
  });

  if (!('base' in locale)) {
    const hydratedChildren = Object.entries(locales)
      .filter(([, childLocale]) => childLocale.base === key)
      .reduce((acc, [childKey, childLocale]) => {
        const mergedLocale = { ...locale, ...childLocale, base: childLocale.base };
        acc[childKey] = buildExpandedLocale(mergedLocale, childKey);
        return acc;
      }, {});

    const hydratedBase = buildExpandedLocale(locale, key);
    return { ...hydratedBase, regions: hydratedChildren };
  }

  if ('base' in locale && locales[locale.base] !== undefined) {
    const baseLocale = locales[locale.base];
    const mergedLocale = { ...baseLocale, ...locale };
    return buildExpandedLocale(mergedLocale, key);
  }
  
  return { ...locale };
}
```